app/cosi/README.md

```diff
@@ -0,0 +1,338 @@
+as a list of conventional signatures from all participants,
+but CoSi signatures are far more compact and efficient for clients to verify.
+In practice, a CoSi collective signature is close to the same size as &mdash;
+and comparable in verification costs to &mdash;
+a *single* individual signature.
+
+CoSi is intended to facilitate increased transparency and security-hardening
+for critical Internet authorities such as certificate authorities,
+[time services](http://www.nist.gov/pml/div688/grp40/its.cfm),
+naming authorities such as [DNSSEC](http://www.dnssec.net),
+software distribution and update services,
+directory services used by tools such as [Tor](https://www.torproject.org),
+and next-generation crypto-currencies.
+For further background and technical details see this research paper:
+* [Keeping Authorities "Honest or Bust" with Decentralized Witness Cosigning](http://dedis.cs.yale.edu/dissent/papers/witness-abs), 
+[IEEE Security & Privacy 2016](http://www.ieee-security.org/TC/SP2016/).
+
+For questions and discussions please join the
+[mailing list](https://groups.google.com/forum/#!forum/cothority).
+
+Other related papers:
+* [Certificate Cothority - Towards Trustworthy Collective CAs](https://petsymposium.org/2015/papers/syta-cc-hotpets2015.pdf), 
+[HotPETS 2015](https://petsymposium.org/2015/hotpets.php)
+* [Enhancing Bitcoin Security and Performance with Strong Consistency via Collective Signing](http://arxiv.org/abs/1602.06997), 
+[USENIX Security 2016](https://www.usenix.org/conference/usenixsecurity16) (to appear)
+ 
+
+**Warning: This software is experimental and still under development.
+Do not use it yet for security-critical purposes.  If you use it,
+do so to supplement, rather than replace, existing signing mechanisms.
+Use at your own risk!**
+
+# Versions
+
+For the moment we have two version: _v0_ 
+(in the [cosi repo](https://github.com/dedis/cosi/)) and 
+_master_ in this repo.
+
+## V0
+
+**Update**: Soon to be deprecated.
+
+This is a stable version that depends on the v0-versions of the other dedis-packages. 
+It will only receive bugfixes, but no changes that will make the code incompatible. 
+You can find this version at:
+
+https://github.com/dedis/cosi/tree/v0
+
+If you write code that uses our library in the v0-version, be sure to reference it as
+
+```
+import "gopkg.in/dedis/cosi.v0"
+```
+
+## Master
+
+The master-branch is used for day-to-day development and will break your 
+code about once a week. If you are using this branch, be sure to do
+
+```
+go get -u -t ./...
+```
+
+from time to time, as all dedis-dependencies change quite often.
+
+# Installation
+
+You may install CoSi from either pre-built binaries
+or from [Go](https://golang.org/) source code,
+as described below.
+
+## Installing from Binaries
+
+For convenience we provide self-contained x86-64 binaries
+for Linux and Mac OS X.
+[Download the latest release](https://github.com/dedis/cosi/releases/latest),
```

Comment:
I'd remove the binary-release for the moment and only keep this section in the v0-branch at `dedis/cosi`.